pkg/operator/ceph/nfs/nfs.go

```diff
@@ -276,6 +300,14 @@ func validateGanesha(context *clusterd.Context, clusterInfo *cephclient.ClusterI
 	// The existence of the pool provided in n.Spec.RADOS.Pool is necessary otherwise addRADOSConfigFile() will fail
 	_, err := cephclient.GetPoolDetails(context, clusterInfo, n.Spec.RADOS.Pool)
 	if err != nil {
+		if strings.Contains(err.Error(), "unrecognized pool") {
+			if clusterInfo.CephVersion.IsAtLeastPacific() {
+				err := createDefaultNFSRADOSPool(context, clusterInfo, n.Spec.RADOS.Pool)
+				if err != nil {
+					return errors.Wrapf(err, "failed to find %q pool and unable to create it", n.Spec.RADOS.Pool)
+				}
```

Comment:
We need to return nil here, right? Or else the reconcile will fail a few lines below when it returns pool not found
```suggestion
				}
				return nil
```